### PR TITLE
Created layout of new model creation wizard page and description wizard page

### PR DESCRIPTION
**Pull request by Yougen-Z**
Added SWTExtensions.

### CODE REVIEW

File: src/testwizard/ModelDescriptionPage.xtend [hunk 1]
```diff
@@ -18,8 +24,36 @@ class ModelDescriptionPage extends WizardPage {
...
+			layout = newGridLayout => [
+                numColumns = 2
+                makeColumnsEqualWidth = false
+            ]
+			addArtifactDescriptionControl
+		]
+		pageComplete = true
 	}
 	
+	def protected void addArtifactDescriptionControl(Composite parent) {
```
Comment (nbhusare) [inline src/testwizard/ModelDescriptionPage.xtend]:
```
parent.addLabel("Description:", SWT.NULL)
parent.layoutData = newGridData() => [
 horizontalAlignment = GridData.BEGINNING
 verticalAlignment = GridData.BEGINNING
]
```

File: src/testwizard/ModelDescriptionPage.xtend [hunk 2]
```diff
@@ -18,8 +24,36 @@ class ModelDescriptionPage extends WizardPage {
...
+        parent.addLabel("Description:", SWT.NULL) => [ 
+            layoutData = newGridData => [
+                horizontalAlignment = GridData.BEGINNING
+                verticalAlignment = GridData.BEGINNING
+            ]
+        ]
+        
+        wizardDesciption = parent.addText(SWT::BORDER.bitwiseOr(SWT::MULTI)) => [
+            layoutData = newGridData => [
```
Comment (nbhusare) [inline src/testwizard/ModelDescriptionPage.xtend]: The layout extension class can have default layout methods that do the following
```
horizontalAlignment = GridData.FILL
verticalAlignment = GridData.FILL
grabExcessHorizontalSpace = true
grabExcessVerticalSpace = true
```

The method could be called - newDefaultGriData() for example

File: src/testwizard/ModelDescriptionPage.xtend [hunk 3]
```diff
@@ -18,8 +24,36 @@ class ModelDescriptionPage extends WizardPage {
 	}
 	
 	override createControl(Composite parent) {
-		// Your code comes here
-		control = parent // You could also assign another composite (created inside this method) to the control. 
+		
+		control = parent.addChildComposite => [
```
Comment (nbhusare) [inline src/testwizard/ModelDescriptionPage.xtend]: ```control = parent.addChildComposite```

Try to use parenthesis at the end of the method cal

``` control = parent.addChildComposite()```

File: src/testwizard/NewModelCreationWizardPage.xtend [hunk 4]
```diff
@@ -2,16 +2,150 @@ package testwizard
...
 import org.eclipse.swt.widgets.Composite
+import org.eclipse.swt.widgets.Button
+import org.eclipse.swt.widgets.Text
+import org.eclipse.swt.SWT
+import static extension com.lib.SWTExtensions.SWTWidgetExtensions.*
+import static extension com.lib.SWTExtensions.SWTLayoutExtensions.*
+import org.eclipse.swt.layout.GridData
 
 class NewModelCreationWizardPage extends WizardPage {
+	Text artifactName
```
Comment (nbhusare) [inline src/testwizard/NewModelCreationWizardPage.xtend]: All the instance variables should be `private`. Make this as a habit while coding.

File: src/testwizard/NewModelCreationWizardPage.xtend [hunk 5]
```diff
@@ -2,16 +2,150 @@ package testwizard
...
 		super("New Model Creation", "New Model Creation", null)
 	}
 
 	override createControl(Composite parent) {
-		// Your code comes here
-		control = parent // You could also assign another composite (created inside this method) to the control. 
+		val ncol = 2 // Number of column in parent Composite
+		
+		control = parent.addChildComposite => [
+            layout = newGridLayout => [
```
Comment (nbhusare) [inline src/testwizard/NewModelCreationWizardPage.xtend]: The lambda appears to be overused. Please see my comments above.

File: src/testwizard/NewModelCreationWizardPage.xtend [hunk 6]
```diff
@@ -2,16 +2,150 @@ package testwizard
...
+    def protected void addArtifactNamespaceDropDown(Composite container) {
+        container.addLabel("Namespace:", SWT.NULL)
+        namespace = container.addText(SWT::BORDER.bitwiseOr(SWT::SINGLE)) => [
+            layoutData = newGridData(GridData::FILL_HORIZONTAL)
+        ]
+    }
+
+    def void addParentSelectionControl(Composite parent) {        
+        parent.addLabel("Parent Model:", SWT::NULL)
+        parent.addChildComposite => [
```
Comment (nbhusare) [inline src/testwizard/NewModelCreationWizardPage.xtend]: Please see comments above

File: src/testwizard/NewModelCreationWizardPage.xtend [hunk 7]
```diff
@@ -2,16 +2,150 @@ package testwizard
...
+    }
+	
+	def void addEntityTypeSelectionControl(Composite parent) {
+        parent.addLabel("Model Type:", SWT::WRAP)        
+        parent.addChildComposite => [
+            layout = newGridLayout => [
+                numColumns = 3
+                makeColumnsEqualWidth = true
+            ]
+            domainModelSelButton = addButton(SWT::RADIO) => [
```
Comment (nbhusare) [inline src/testwizard/NewModelCreationWizardPage.xtend]: Please see comments above

File: src/testwizard/NewModelCreationWizardPage.xtend [hunk 8]
```diff
@@ -2,16 +2,150 @@ package testwizard
...
+    }
+
+    def void addModifierControls(Composite parent) {
+        parent.addLabel("Modefiers:", SWT::NONE)
+        parent.addChildComposite(SWT::NULL) => [
+            layout = newGridLayout => [
+                numColumns = 3
+                makeColumnsEqualWidth = true
+            ]
+            abstractSelButton = addButton(SWT::CHECK) => [
```
Comment (nbhusare) [inline src/testwizard/NewModelCreationWizardPage.xtend]: Please see comments above

File: src/testwizard/NewModelCreationWizardPage.xtend [hunk 9]
```diff
@@ -2,16 +2,150 @@ package testwizard
...
+            nonCacheableSelButton = addButton(SWT::CHECK) => [
+                text = "NonCacheable"
+            ]
+        ] 
+    }
+	
+    def protected void addOpenInEditorCheckbox(Composite parent) {
+    	parent.addChildComposite(SWT::NULL) => [
+			layout = newGridLayout
+	        openInEditorChkBox = parent.addButton(SWT::CHECK) => [
```
Comment (nbhusare) [inline src/testwizard/NewModelCreationWizardPage.xtend]: Please see comments above

File: src/testwizard/NewModelCreationWizardPage.xtend [hunk 10]
```diff
@@ -2,16 +2,150 @@ package testwizard
...
+        ]
+    }
+	
+	def void createLine(Composite parent, int ncol) 
+	{
+		parent.addLabel(SWT::SEPARATOR
+			.bitwiseOr(SWT::HORIZONTAL)
+			.bitwiseOr(SWT::BOLD)) => [
+				
+			layoutData = newGridData(GridData.FILL_HORIZONTAL) => [
```
Comment (nbhusare) [inline src/testwizard/NewModelCreationWizardPage.xtend]: Try not to nest the lambdas too much. You lose the readability

File: src/handlers/NewModelCreationWizardHandler.xtend [hunk 11]
```diff
@@ -0,0 +1,11 @@
+package handlers
+
+import testwizard.NewModelCreationWizard
+
+class NewModelCreationWizardHandler extends AbstractArtifactWizardHandler {
+	
+	override getArtifactCreationWizard() {
+		new NewModelCreationWizard
```
Comment (nbhusare) [inline src/handlers/NewModelCreationWizardHandler.xtend]: Please use parenthesis at the end of the method.

File: src/com/lib/SWTExtensions/SWTLayoutExtensions.xtend [hunk 12]
```diff
@@ -0,0 +1,24 @@
+package com.lib.SWTExtensions
+
+import org.eclipse.swt.layout.GridLayout
+import org.eclipse.swt.layout.GridData
+import org.eclipse.swt.layout.FillLayout
+
+class SWTLayoutExtensions {
```
Comment (nbhusare) [inline src/com/lib/SWTExtensions/SWTLayoutExtensions.xtend]: Add default method that does the following
```
horizontalAlignment = GridData.FILL
verticalAlignment = GridData.FILL
grabExcessHorizontalSpace = true
grabExcessVerticalSpace = true
```

**Comment by nbhusare** [comment]
Can you please attach some screen shots of the wizard

File: src/testwizard/ModelDescriptionPage.xtend (hunk 3 above)
Comment (Yougen-Z) [inline src/testwizard/ModelDescriptionPage.xtend]: Fixed, Thanks!

Comment (Yougen-Z) [inline src/testwizard/ModelDescriptionPage.xtend]: Fixed, Thanks!

File: src/testwizard/ModelDescriptionPage.xtend (hunk 1 above)
Comment (Yougen-Z) [inline src/testwizard/ModelDescriptionPage.xtend]: Fixed, Thanks!

File: src/testwizard/ModelDescriptionPage.xtend (hunk 2 above)
Comment (Yougen-Z) [inline src/testwizard/ModelDescriptionPage.xtend]: Fixed, Thanks!

File: src/testwizard/NewModelCreationWizardPage.xtend (hunk 4 above)
Comment (Yougen-Z) [inline src/testwizard/NewModelCreationWizardPage.xtend]: Fixed, Thanks!

File: src/testwizard/NewModelCreationWizardPage.xtend (hunk 5 above)
Comment (Yougen-Z) [inline src/testwizard/NewModelCreationWizardPage.xtend]: Fixed, Thanks!

File: src/testwizard/NewModelCreationWizardPage.xtend (hunk 6 above)
Comment (Yougen-Z) [inline src/testwizard/NewModelCreationWizardPage.xtend]: Fixed, Thanks!

File: src/testwizard/NewModelCreationWizardPage.xtend (hunk 8 above)
Comment (Yougen-Z) [inline src/testwizard/NewModelCreationWizardPage.xtend]: Fixed, Thanks!

File: src/testwizard/NewModelCreationWizardPage.xtend (hunk 9 above)
Comment (Yougen-Z) [inline src/testwizard/NewModelCreationWizardPage.xtend]: Fixed, Thanks!

File: src/com/lib/SWTExtensions/SWTLayoutExtensions.xtend (hunk 12 above)
Comment (Yougen-Z) [inline src/com/lib/SWTExtensions/SWTLayoutExtensions.xtend]: Fixed, Thanks!

File: src/handlers/NewModelCreationWizardHandler.xtend (hunk 11 above)
Comment (Yougen-Z) [inline src/handlers/NewModelCreationWizardHandler.xtend]: Fixed, Thanks!

**Comment by Yougen-Z** [comment]
model wizard page
description page

**Pull request merged**